provisioning/device/samples/getting started/X509Sample/ProvisioningDeviceClientSample.cs

```diff
@@ -16,76 +18,110 @@ namespace Microsoft.Azure.Devices.Provisioning.Client.Samples
     /// </summary>
     internal class ProvisioningDeviceClientSample
     {
-        private readonly Parameters _parameters;
-
-        public ProvisioningDeviceClientSample(Parameters parameters)
+        private readonly ILogger _logger;
+        private static volatile ProvisioningDeviceClient s_provClient;
+        private static volatile DeviceRegistrationResult s_provResult;
+        private static ProvisioningClientOptions s_clientOptions;
+        private static IotHubClientTransportSettings s_hubTransportSettings;
+        private static CancellationTokenSource s_appCancellation;
+        private readonly string _idScope;
+        private readonly string _globalDeviceEndpoint;
+        private string _certificatePassword;
+        private readonly string _certificatePath;
+        private readonly string _certificateName;
+
+        public ProvisioningDeviceClientSample(Parameters parameters, ILogger logger)
         {
-            _parameters = parameters;
+            s_clientOptions = parameters.GetClientOptions();
+            s_hubTransportSettings = parameters.GetHubTransportSettings();
+            _certificatePath = parameters.GetCertificatePath();
+            _logger = logger;
+            _idScope = parameters.IdScope;
+            _globalDeviceEndpoint = parameters.GlobalDeviceEndpoint;
+            _certificateName = parameters.CertificateName;
+            _certificatePassword = parameters.CertificatePassword;
         }
 
         public async Task RunSampleAsync()
         {
-            Console.WriteLine($"Loading the certificate...");
+            s_appCancellation = new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                s_appCancellation.Cancel();
+                _logger.LogWarning("Sample execution cancellation requested; will exit.");
+            };
+
+            _logger.LogInformation("Loading the certificate...");
             using X509Certificate2 certificate = LoadProvisioningCertificate();
             var security = new AuthenticationProviderX509(certificate);
 
-            Console.WriteLine($"Initializing the device provisioning client...");
+            // Perform retry up to 10 times every 2 seconds
+            s_clientOptions.RetryPolicy = new ProvisioningClientFixedDelayRetryPolicy(10, TimeSpan.FromSeconds(2));
```

Comment:
Should we have parameters for the retry policy values instead?

Reply:
Make sense for getting started to use a simple fixed delay retry policy. But as an example, maybe instead of commenting "Perform retry frequency", it would be more insightful to comment there are more advanced exponential backoff retry policy in the SDK.

Reply:
Sure, I can make them configurable.

Reply:
Exponential backoff retry policy is our default retry policy. Should I use exponential backoff retry policy instead?

Reply:
I would think there are 2 approaches:
1) use fixed retry policy (as custom) to show how to do custom, but note that this is to show how to do custom retry and not recommending fix delay retry for production.
2) use default exp retry policy to show best practice, but not that customer can create custom policy to accommodate specific needs.

Either way, I was thinking there should be additional comment to explain this in the sample.